### ISSUE TITLE
Carousel: Empty heading tags are created

### ISSUE THREAD

**Issue by mrekici** [Automattic/jetpack#29141]
### Impacted plugin

Jetpack

### What

Empty heading tags

### How

I am using the jetpack carousel. After loading the page, before images are selected, jetpack generates empty heading tags:

```
<h2 class="jp-carousel-photo-title"></h2>
<h2 class="jp-carousel-photo-caption"></h2>
<h3 class="jp-carousel-caption" itemprop="caption description"></h3>
```

How can I turn this off so that no empty heading tags are generated?

Thanks.

**Comment by jeherve** [comment]
Those are indeed added to the markup on page load, but they should then be populated when the page finishes loading. Do you not see any text inside the heading on your end? If could, could you post a URL where you see the issue here, so we can take a closer look?

Thank you!

**Comment by mrekici** [comment]
Thanks for the reply. The issue is that I don't see the headings, as they are generated without a command. 

Website: https://emreekici.com/about-me-sound-engineer/

Example:
`<h2 class="jp-carousel-photo-title"></h2>`

The issue is a problem for me as these are considered headings for SEO, and there is no heading and no text under the heading.

Thanks again for the help!

**Comment by jeherve** [comment]
Thanks for the extra details!

There seems to be something wrong on your site at the moment. I think that some of your caching and concatenating options[^1] are causing lots of JavaScript files to not be loaded properly on your site:

<img width="1376" alt="image" src="https://user-images.githubusercontent.com/426388/221949274-49995af0-9f21-453a-a7e1-260274a4ade9.png">

This, in turn, breaks Jetpack's Carousel feature. The headings should be injected via JavaScript into the Carousel view when the page finishes loading, but since JavaScript files are broken on your site, this doesn't happen.
Once you fix that issue, you should see the headings appear as planned.

Give it a try, and let me know how it goes!

[^1]: You may be able to disable your caching plugin. Since your site is hosted on WordPress.com, it already comes with its own caching solution. [You can read more about it here](https://wordpress.com/support/plugins/incompatible-plugins/#caching-plugins).

**Comment by bugrevealingbme** [comment]
Yes, this problem can be harmful to SEO. We don't want Jetpack to use H2 and H3 here.

You can also visit this issue on my page.
https://xiaomiui.net/redmi-13c-spotted-on-gsma-imei-database-51567/

**Comment by stronenv** [comment]
We're seeing these tags loaded on a page with no carousel, so no content is added to the tatgs - is that intentional? 

I've turned off the Jetpack Carousel on the site in question since we're not using it, but let me know if you'd like me to provide an example.

**Comment by jeherve** [comment]
@stronenv If you can give me an example that would indeed be helpful, thank you! Those tags should only be added when using the Carousel.

**Comment by stronenv** [comment]
Hey @jeherve! 

Sure, I was able to reproduce it on a fresh site: https://stronenvtestingjpcarousel.wpcomstaging.com/

^ That's on WordPress.com, which might be relevant.

**Comment by jeherve** [comment]
@stronenv I see it now, thank you!

**Comment by kathrynwp** [comment]
@jeherve User in 7438445-zd-a8c is also observing Carousel code loading on pages which **do not** have a carousel, including empty heading tags, which they are concerned about for SEO reasons. 

I advised user to turn off image carousels in Jetpack settings to remove the extra code.

**Comment by rickmgithub** [comment]
The issue was observed when checking through the WAVE tool - https://wave.webaim.org/report#/helendentwrites.com

8419082-zd-a8c

**Comment by rinazrina** [comment]
Another report: 9617475-zd-a8c

**Comment by jromales** [comment]
Adding another report: 11060290-zd-a8c
